explorer/server.go

```diff
@@ -0,0 +1,210 @@
+package explorer
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/gomodule/redigo/redis"
+	"github.com/gorilla/mux"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog/log"
+	"github.com/threefoldtech/zos/client"
+	"github.com/threefoldtech/zos/pkg/rmb"
+)
+
+func (a *App) listFarms(w http.ResponseWriter, r *http.Request) {
+	r, err := a.handleRequestsQueryParams(r)
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write([]byte(http.StatusText(http.StatusBadRequest)))
+		return
+	}
+	maxResult, pageOffset := getMaxResult(r.Context()), getOffset(r.Context())
+
+	queryString := fmt.Sprintf(`
+	{
+		farms (limit:%d,offset:%d) {
+			name
+			farmId
+			twinId
+			version
+			farmId
+			pricingPolicyId
+		}
+		publicIps{
+			id
+			ip
+			farmId
+			contractId
+			gateway
+			
+		}
+	}
+	`, maxResult, pageOffset)
+
+	_, err = queryProxy(queryString, a.explorer, w)
+
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
```

Comment:
this should be `internal server error` (also in listNodes)? the `handleRequestsQueryParams` seems to be doing the user input validation, but if `queryProxy` failed, then it's either a connection error or interoperability issue, which are internal server errors.